### PR TITLE
LPS-152905 Add private method to retrieve classNameId

### PR DESCRIPTION
**Pull request by Stephy6**

### CODE REVIEW

**Comment by Stephy6** [comment]
Hi @cgoncas ! I have doubts in the doServeResource method.  There are two variables classPK and className that seems that we're reusing but I'm not sure. Could you take a look? 
Thank you.

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import com.liferay.asset.kernel.model.AssetVocabulary;
 import com.liferay.asset.kernel.model.AssetVocabularyConstants;
 import com.liferay.asset.kernel.service.AssetVocabularyLocalService;
+import com.liferay.blogs.model.BlogsEntry;
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:22]: Content Dashboard API should never depend in any asset, so we can create a dependency with blogs

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java [hunk 2]
```diff
@@ -126,17 +132,29 @@ protected void doServeResource(
 					}
 				}
 			).map(
-				contentDashboardItem -> JSONUtil.put(
+				contentDashboardItem -> {
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:135]: Add a new line to put _getClassNameId.

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java [hunk 3]
```diff
@@ -309,6 +334,39 @@ private JSONObject _getClipboardJSONObject(
...
+				themeDisplay.getSiteGroup());
+
+		if (!sharingConfiguration.isEnabled()) {
+			return false;
+		}
+
+		boolean sharingEnabled = false;
+
+		try {
+			sharingEnabled = _sharingPermission.containsSharePermission(
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:352]: return fast, instead save the result in a variable.

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java [hunk 4]
```diff
@@ -309,6 +334,39 @@ private JSONObject _getClipboardJSONObject(
 			});
 	}
 
+	private boolean _isSharingEnabled(
+		long classNameId, long classPK, ContentDashboardItem<?> contentDashboardItem,
+		ThemeDisplay themeDisplay ) {
+
+		SharingConfiguration sharingConfiguration =
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:341]: You should read the company segmentation too

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java (hunk 1 above)
Comment (Stephy6) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:22]: Removed! thank you!

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java [hunk 5]
```diff
@@ -47,6 +49,7 @@
 import com.liferay.portal.kernel.util.LocaleUtil;
 import com.liferay.portal.kernel.util.ParamUtil;
 import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.PortalUtil;
```
[the remark this answers is not in the document]
Comment (Stephy6) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:52]: Removed! thank you

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java (hunk 2 above)
Comment (Stephy6) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:134]: Done! thank you!

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java (hunk 3 above)
Comment (Stephy6) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java]: Done! thank you!

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java (hunk 4 above)
Comment (Stephy6) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java]: Done! thank you!

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java [hunk 6]
```diff
@@ -432,6 +462,54 @@ private JSONArray _getViewURLsJSONArray(
...
+
+		SharingConfiguration companySharingConfiguration =
+			_sharingConfigurationFactory.getCompanySharingConfiguration(
+				company);
+
+		if (!companySharingConfiguration.isEnabled()) {
+			return false;
+		}
+
+		SharingConfiguration sharingConfiguration =
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:488]: Maybe I am wrong, but I think taking a look to getGroupSharingConfiguration, that it checked getCompanySharingConfiguration too.

Also, maybe would be easy to find where all this logic is used and create an API to share this logic

**Comment by cgoncas** [comment]
> Checking for the code of the Sharing Permission, I just have found this API: SharingMenuItemFactory, that returns a MenuItem, It returns a JavaScriptMenuItem, maybe we can use it in order to paint the Share Icon?

Even better, we have the SharingJavaScriptFactory::createSharingOnClickMethod that returns the right code.

**Comment by claraizquierdo** [comment]
> Yeahh!! Maybe it is clearer, but this is the code that is used in all the DL, so if someone changes something, we won’t need to make any changes

Ok, no problem. I can change it :) Now we will not need the classNameId in the front, just the keys `shareable` and `onShareClick` right?

**Comment by beltranrengifo** [comment]
@cgoncas @claraizquierdo 
I understand the benefits of reusing `SharingJavaScriptFactory::createSharingOnClickMethod` however this method is returning a string representing the execution of the function, not its name. This would force the FE to execute the expression in a _better-to-avoid_ way. [Check line 17](https://codesandbox.io/s/red-flower-blhc3i?file=/src/App.js)
Not a big deal but in the future could be great to explore a more modern way of dealing with this. 
What do you think?

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java [hunk 7]
```diff
@@ -432,6 +470,54 @@ private JSONArray _getViewURLsJSONArray(
 			).toArray());
 	}
 
+	private boolean _isShareable(
```
Comment (claraizquierdo) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:473]: I don't know if we are checking that the item is a document or a blog in order to be shareable. Right now I'm able to see the share button in Web Content Articles and I shouldn't.

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java [hunk 8]
```diff
@@ -502,6 +506,15 @@ private boolean _isShareable(
 		}
 
 		try {
+			String className = _portal.getClassName(classNameId);
+
+			// LPS-111037
+
+			if (Objects.equals(FileEntry.class.getName(), className)) {
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:513]: Hey @Stephy6 !!

As it hapends with Blogs, we should never add a explicit dependency to files in the Content Dashboard, take a look into the actual code to see how we are mapping betwen FileEntry and DLFileEntry, for example when we use Search in the Content Dashboard

**Comment by cgoncas** [comment]
Hey guys!!

i just created the PR: https://github.com/liferay-tango/liferay-portal/pull/2612, with a new JSP that shows the share button.

I create another PR, so we can keep this code too and decide what is the best solution.

@claraizquierdo , could you add the code to render the Share button in the info panel with the given URL?

Let me know what do you think.

See you!!

**Pull request closed without merging**